src/backend/distributed/sql/udfs/citus_job_status/latest.sql

```diff
@@ -0,0 +1,73 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_job_status (
+    job_id bigint DEFAULT NULL
+)
+    RETURNS TABLE (
+            job_id bigint,
+            state pg_catalog.citus_job_status,
+            job_type name,
+            description text,
+            started_at timestamptz,
+            finished_at timestamptz,
+            details jsonb
+    )
+    LANGUAGE SQL
+    STRICT
+    AS $fn$
+    WITH rp AS (
+        SELECT * FROM get_rebalance_progress()
+    ),
+    task_state_occurence_counts AS (
+        SELECT t.status, count(task_id)
+        FROM pg_dist_background_job j
+            JOIN pg_dist_background_task t ON t.job_id = j.job_id
+        WHERE j.job_id = $1
+        GROUP BY t.status
+    ),
+    running_task_details AS (
+        SELECT jsonb_agg(jsonb_build_object(
+                'state', t.status,
+                'retried', coalesce(t.retry_count,0),
+                'phase', rp.status,
+                'size' , jsonb_build_object(
+                    'source', rp.source_shard_size,
+                    'target', rp.target_shard_size),
+                'hosts', jsonb_build_object(
+                    'source', rp.sourcename || ':' || rp.sourceport,
+                    'target', rp.targetname || ':' || rp.targetport),
+                'message', t.message,
+                'command', t.command,
+                'LSN', jsonb_build_object(
+                    'source', rp.source_lsn,
+                    'target', rp.target_lsn),
+                'task_id', t.task_id )) AS tasks
+        FROM
+            rp JOIN pg_dist_background_task t ON rp.sessionid = t.pid
```

Comment:
I think a filter on the `running` state would be good here.

Reply:
Now I have 2 table expressions, one for running state and one for error state.